.github/workflows/bindings_python.yml

```diff
@@ -0,0 +1,273 @@
+            pip install imap-codec --find-links ../../target/wheels --force-reinstall
+            pytest
+
+  windows:
+    runs-on: ${{ matrix.platform.runner }}
+    strategy:
+      matrix:
+        platform:
+          - runner: windows-latest
+            target: x64
+          - runner: windows-latest
+            target: x86
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+          architecture: ${{ matrix.platform.target }}
+      - name: Build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.platform.target }}
+          args: --release --find-interpreter
+          sccache: 'true'
+          working-directory: bindings/imap-codec-python
+      - name: Upload wheels
+        uses: actions/upload-artifact@v4
+        with:
+          name: wheels-windows-${{ matrix.platform.target }}
+          path: target/wheels
+      - name: pytest
+        if: ${{ !startsWith(matrix.platform.target, 'aarch64') }}
+        shell: bash
+        run: |
+          set -e
+          cd bindings/imap-codec-python
+          python3 -m venv venv
+          source venv/Scripts/activate
+          pip install imap-codec --find-links ../../target/wheels --force-reinstall
+          pip install pytest
+          pytest
+
+  macos:
+    runs-on: ${{ matrix.platform.runner }}
+    strategy:
+      matrix:
+        platform:
+          - runner: macos-12
+            target: x86_64
+          - runner: macos-14
+            target: aarch64
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+      - name: Build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.platform.target }}
+          args: --release --find-interpreter
+          sccache: 'true'
+          working-directory: bindings/imap-codec-python
+      - name: Upload wheels
+        uses: actions/upload-artifact@v4
+        with:
+          name: wheels-macos-${{ matrix.platform.target }}
+          path: target/wheels
+      - name: pytest
+        run: |
+          set -e
+          cd bindings/imap-codec-python
+          python3 -m venv venv
+          source venv/bin/activate
+          pip install imap-codec --find-links ../../target/wheels --force-reinstall
+          pip install pytest
+          pytest
+
+  sdist:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build sdist
+        uses: PyO3/maturin-action@v1
+        with:
+          command: sdist
+          working-directory: bindings/imap-codec-python
+      - name: Upload sdist
+        uses: actions/upload-artifact@v4
+        with:
+          name: wheels-sdist
+          path: target/wheels
+
+  release:
+    name: Release package to PyPI
+    runs-on: ubuntu-latest
+    if: ${{ startsWith(github.ref, 'refs/tags/imap-codec-python/v') }}
```

Comment:
Okay, so tags are `imap-codec-python/vx.y.z` :+1:

We could also remove the `${{ ` and ` }}` to reduce the diff from the maturin template.

Reply:
The VS Code extension for GitHub Actions did not like that syntax, thus, I thought this is deprecated syntax.

However, that seems to be a bug in the extension, as the [documentation](https://docs.github.com/en/actions/learn-github-actions/expressions#about-expressions) states:
> Note: The exception to this rule is when you are using expressions in an `if` clause, where, optionally, you can usually omit `${{` and `}}`. 

I'll go back to using double quotes then.